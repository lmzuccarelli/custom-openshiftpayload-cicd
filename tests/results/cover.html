
<!DOCTYPE html>
<html>
	<head>
		<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
		<title>service: Go Coverage Report</title>
		<style>
			body {
				background: black;
				color: rgb(80, 80, 80);
			}
			body, pre, #legend span {
				font-family: Menlo, monospace;
				font-weight: bold;
			}
			#topbar {
				background: black;
				position: fixed;
				top: 0; left: 0; right: 0;
				height: 42px;
				border-bottom: 1px solid rgb(80, 80, 80);
			}
			#content {
				margin-top: 50px;
			}
			#nav, #legend {
				float: left;
				margin-left: 10px;
			}
			#legend {
				margin-top: 12px;
			}
			#nav {
				margin-top: 10px;
			}
			#legend span {
				margin: 0 5px;
			}
			.cov0 { color: rgb(192, 0, 0) }
.cov1 { color: rgb(128, 128, 128) }
.cov2 { color: rgb(116, 140, 131) }
.cov3 { color: rgb(104, 152, 134) }
.cov4 { color: rgb(92, 164, 137) }
.cov5 { color: rgb(80, 176, 140) }
.cov6 { color: rgb(68, 188, 143) }
.cov7 { color: rgb(56, 200, 146) }
.cov8 { color: rgb(44, 212, 149) }
.cov9 { color: rgb(32, 224, 152) }
.cov10 { color: rgb(20, 236, 155) }

		</style>
	</head>
	<body>
		<div id="topbar">
			<div id="nav">
				<select id="files">
				
				<option value="file0">github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go (87.9%)</option>
				
				<option value="file1">github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go (82.0%)</option>
				
				</select>
			</div>
			<div id="legend">
				<span>not tracked</span>
			
				<span class="cov0">not covered</span>
				<span class="cov8">covered</span>
			
			</div>
		</div>
		<div id="content">
		
		<pre class="file" id="file0" style="display: none">package service

import (
        "bytes"
        "fmt"
        "io/ioutil"
        "path/filepath"
        "reflect"
        "strings"
        "text/template"

        "github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/schema"
        "gopkg.in/yaml.v2"
)

const (
        build           string = "Build"
        buildConfig     string = "BuildConfig"
        manifests       string = "/manifests/"
        dotYml          string = ".yaml"
        errMsgYaml      string = "converting struct to yaml for %s"
        errMsgUnmarshal string = "unmarshaling yaml to struct %v for %s"
)

var taskRunTemplate = `
kind: TaskRun
apiVersion: 0.0.1
metadata:
  name: {{ .Name }}
spec:
  taskRef:
    name: {{ .TaskRef }}
  params:
    - name: repository-url
      value: '{{ .GitURL }}'
    - name: repository-name
      value: '{{ .Name }}'
    - name: branch
      value: 'master'
    - name: dockerfile
      value: '{{ .Dockerfile }}'
`

func deepCopyTask(task schema.Task) schema.Task <span class="cov8" title="1">{
        t := schema.Task{}
        t.Kind = task.Kind
        t.APIVersion = task.APIVersion
        t.Metadata.Name = task.Metadata.Name
        s := schema.Step{}
        for _, step := range task.Spec.Steps </span><span class="cov8" title="1">{
                s.Workspace = step.Workspace
                cmd := []string{}
                args := []string{}
                s.Name = step.Name
                s.Description = step.Description
                for _, s := range step.Command </span><span class="cov8" title="1">{
                        cmd = append(cmd, s)
                }</span>
                <span class="cov8" title="1">for _, a := range step.Args </span><span class="cov8" title="1">{
                        args = append(args, a)
                }</span>
                <span class="cov8" title="1">s.Command = cmd
                s.Args = args
                t.Spec.Steps = append(t.Spec.Steps, s)</span>
        }
        <span class="cov8" title="1">return t</span>
}

func mergeParams(tr *schema.TaskRun, pt *schema.Pipeline) <span class="cov8" title="1">{
        for _, p := range pt.Spec.Params </span><span class="cov8" title="1">{
                tr.Spec.Params = append(tr.Spec.Params, schema.Param{Name: p.Name, Value: p.Default})
        }</span>
}

func updateParameters(task schema.Task, taskrun schema.TaskRun) <span class="cov8" title="1">{
        for x, step := range task.Spec.Steps </span><span class="cov8" title="1">{
                for y, arg := range step.Args </span><span class="cov8" title="1">{
                        if strings.Contains(arg, "${params.") </span><span class="cov8" title="1">{
                                task.Spec.Steps[x].Args[y] = replaceWithParamValue(taskrun, arg)
                        }</span>
                }
                <span class="cov8" title="1">if strings.Contains(step.Workspace, "${params.") </span><span class="cov0" title="0">{
                        task.Spec.Steps[x].Workspace = replaceWithParamValue(taskrun, step.Workspace)
                }</span>
        }
}

func replaceWithParamValue(task schema.TaskRun, value string) string <span class="cov8" title="1">{
        newValue := value
        for _, param := range task.Spec.Params </span><span class="cov8" title="1">{
                newValue = strings.ReplaceAll(newValue, "${params."+param.Name+"}", param.Value)
        }</span>
        <span class="cov8" title="1">return newValue</span>
}

func findRelatedTask(tasks []schema.Task, reference string) schema.Task <span class="cov8" title="1">{
        for _, task := range tasks </span><span class="cov8" title="1">{
                if reference == task.Metadata.Name </span><span class="cov8" title="1">{
                        return task
                }</span>
        }
        <span class="cov0" title="0">return schema.Task{}</span>
}

func readAllTaskFiles(dir string, files []string) ([]schema.Task, error) <span class="cov8" title="1">{
        var tasks []schema.Task
        for _, f := range files </span><span class="cov8" title="1">{
                hldTask, err := yamlToStruct(dir+"/"+f, schema.Task{})
                t, ok := hldTask.(schema.Task)
                if err != nil &amp;&amp; ok </span><span class="cov0" title="0">{
                        return []schema.Task{}, err
                }</span>
                <span class="cov8" title="1">tasks = append(tasks, t)</span>
        }
        <span class="cov8" title="1">return tasks, nil</span>
}

func readAllTaskRunFiles(dir string, files []string) ([]schema.TaskRun, error) <span class="cov8" title="1">{
        var taskruns []schema.TaskRun
        var tr schema.TaskRun
        var ok bool

        for _, f := range files </span><span class="cov8" title="1">{
                hldTr, err := yamlToStruct(dir+"/"+f, schema.TaskRun{})
                tr, ok = hldTr.(schema.TaskRun)
                if err != nil &amp;&amp; !ok </span><span class="cov0" title="0">{
                        return []schema.TaskRun{}, err
                }</span>
                <span class="cov8" title="1">taskruns = append(taskruns, tr)</span>
        }
        <span class="cov8" title="1">return taskruns, nil</span>
}

// readAllBuildConfigs - reads all the BuildConfigs from a given directory
func readAllBuildConfigs(dir string) ([]schema.BuildConfig, error) <span class="cov8" title="1">{
        var bcs []schema.BuildConfig
        files, err := ioutil.ReadDir(dir)
        if err != nil </span><span class="cov0" title="0">{
                return []schema.BuildConfig{}, err
        }</span>
        <span class="cov8" title="1">for _, f := range files </span><span class="cov8" title="1">{
                if !f.IsDir() </span><span class="cov8" title="1">{
                        hldBC, err := yamlToStruct(dir+"/"+f.Name(), schema.BuildConfig{})
                        bc, ok := hldBC.(schema.BuildConfig)
                        if err != nil &amp;&amp; ok </span><span class="cov0" title="0">{
                                return []schema.BuildConfig{}, err
                        }</span>
                        // we are only interested in BuildCobcnfigs
                        <span class="cov8" title="1">if bc.Kind == buildConfig </span><span class="cov8" title="1">{
                                bcs = append(bcs, bc)
                        }</span>
                }
        }
        <span class="cov8" title="1">return bcs, nil</span>
}

func generateTaskRunFiles(dir string, bcs []schema.BuildConfig) error <span class="cov8" title="1">{
        for _, bc := range bcs </span><span class="cov8" title="1">{
                schema := &amp;schema.TemplateSchema{Name: filepath.Base(bc.Spec.Source.Git.URI), GitURL: bc.Spec.Source.Git.URI, Dockerfile: bc.Spec.Strategy.DockerStrategy.DockerfilePath, TaskRef: "custom-openshift-build"}
                //parse some content and generate a template
                tmpl := template.New("taskruns")
                tmp, _ := tmpl.Parse(taskRunTemplate)
                var tpl bytes.Buffer
                tmp.Execute(&amp;tpl, schema)
                err := ioutil.WriteFile(dir+"/"+schema.Name+".yaml", tpl.Bytes(), 0755)
                if err != nil </span><span class="cov0" title="0">{
                        return err
                }</span>
        }
        <span class="cov8" title="1">return nil</span>
}

func yamlToStruct(file string, strctType interface{}) (interface{}, error) <span class="cov8" title="1">{
        yfile, err := ioutil.ReadFile(file)
        if err != nil </span><span class="cov8" title="1">{
                return strctType, err
        }</span>
        <span class="cov8" title="1">v := reflect.TypeOf(strctType).String()
        switch v </span>{
        case "schema.BuildConfig":<span class="cov8" title="1">
                var bc schema.BuildConfig
                err = yaml.Unmarshal(yfile, &amp;bc)
                if err != nil </span><span class="cov0" title="0">{
                        return strctType, fmt.Errorf(errMsgUnmarshal, err, file)
                }</span>
                <span class="cov8" title="1">return bc, nil</span>
        case "schema.TaskRun":<span class="cov8" title="1">
                var tr schema.TaskRun
                err = yaml.Unmarshal(yfile, &amp;tr)
                if err != nil </span><span class="cov0" title="0">{
                        return strctType, fmt.Errorf(errMsgUnmarshal, err, file)
                }</span>
                <span class="cov8" title="1">return tr, nil</span>
        case "schema.Task":<span class="cov8" title="1">
                var t schema.Task
                err = yaml.Unmarshal(yfile, &amp;t)
                if err != nil </span><span class="cov0" title="0">{
                        return strctType, fmt.Errorf(errMsgUnmarshal, err, file)
                }</span>
                <span class="cov8" title="1">return t, nil</span>
        case "schema.Pipeline":<span class="cov8" title="1">
                var p schema.Pipeline
                err = yaml.Unmarshal(yfile, &amp;p)
                if err != nil </span><span class="cov0" title="0">{
                        return strctType, fmt.Errorf(errMsgUnmarshal, err, file)
                }</span>
                <span class="cov8" title="1">return p, nil</span>
        case "schema.Kustomization":<span class="cov8" title="1">
                var k schema.Kustomization
                err = yaml.Unmarshal(yfile, &amp;k)
                if err != nil </span><span class="cov0" title="0">{
                        return strctType, fmt.Errorf(errMsgUnmarshal, err, file)
                }</span>
                <span class="cov8" title="1">return k, nil</span>
        }
        <span class="cov0" title="0">return strctType, nil</span>
}
</pre>
		
		<pre class="file" id="file1" style="display: none">package service

import (
        "fmt"
        "os"
        "reflect"
        "strings"

        "github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/connectors"
        "github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/schema"
)

func ExecutePipeline(path string, c connectors.Clients) error <span class="cov8" title="1">{

        var p schema.Pipeline
        var t []schema.Task
        var tr []schema.TaskRun
        var ok bool

        os.Chdir(path)
        // read the main kustomization yaml file
        hldK, err := yamlToStruct("./kustomization.yaml", schema.Kustomization{})
        k, ok := hldK.(schema.Kustomization)
        if err != nil &amp;&amp; !ok </span><span class="cov0" title="0">{
                return err
        }</span>
        <span class="cov8" title="1">c.Trace("main kustomization file : %v", k)

        // get all the relevant files to load
        for _, f := range k.Bases </span><span class="cov8" title="1">{
                os.Chdir(f)
                subK, err := yamlToStruct("./kustomization.yaml", schema.Kustomization{})
                k, ok := subK.(schema.Kustomization)
                if err != nil &amp;&amp; !ok </span><span class="cov0" title="0">{
                        return err
                }</span>
                <span class="cov8" title="1">c.Trace("sub level kustomization file %s : %v", f, subK)

                switch </span>{
                case strings.Contains(f, "pipelines"):<span class="cov8" title="1">
                        // only the first pipeline file will be used
                        // other pipleine files will be ignored
                        hldPt, err := yamlToStruct(f+"/"+k.Bases[0], schema.Pipeline{})
                        p, ok = hldPt.(schema.Pipeline)
                        if err != nil &amp;&amp; !ok </span><span class="cov0" title="0">{
                                return err
                        }</span>
                        <span class="cov8" title="1">c.Debug("pipelines : %v", p)
                        if len(p.Spec.Workspaces) == 0 </span><span class="cov8" title="1">{
                                c.Info("field 'Workspaces is empty current directory 'working-dir' is set as default")
                                p.Spec.Workspaces = append(p.Spec.Workspaces, schema.Workspace{Name: "./working-dir"})
                        }</span>
                        <span class="cov8" title="1">c.Info("deleting working directory %s", p.Spec.Workspaces[0].Name)
                        err = os.RemoveAll(p.Spec.Workspaces[0].Name)
                        if err != nil </span><span class="cov0" title="0">{
                                return err
                        }</span>
                        <span class="cov8" title="1">c.Info("creating working directory %s", p.Spec.Workspaces[0].Name)
                        err = os.MkdirAll(p.Spec.Workspaces[0].Name, os.ModePerm)
                        if err != nil </span><span class="cov0" title="0">{
                                return err
                        }</span>
                case strings.Contains(f, "tasks"):<span class="cov8" title="1">
                        t, err = readAllTaskFiles(f, k.Bases)
                        if err != nil </span><span class="cov0" title="0">{
                                return err
                        }</span>
                        <span class="cov8" title="1">c.Debug("tasks : %v", t)</span>
                case strings.Contains(f, "taskruns"):<span class="cov8" title="1">
                        tr, err = readAllTaskRunFiles(f, k.Bases)
                        if err != nil </span><span class="cov0" title="0">{
                                return err
                        }</span>
                        <span class="cov8" title="1">c.Debug("taskruns : %v", tr)
                        if len(tr) == 0 </span><span class="cov0" title="0">{
                                return fmt.Errorf("taskrun list is empty (verify taskruns kustomization file)")
                        }</span>
                }
        }

        // we now execute the pipeline from the taskruns
        <span class="cov8" title="1">for _, taskrun := range tr </span><span class="cov8" title="1">{
                mergeParams(&amp;taskrun, &amp;p)
                task := findRelatedTask(t, taskrun.Spec.TaskRef.Name)
                if reflect.DeepEqual(task, schema.Task{}) </span><span class="cov0" title="0">{
                        return fmt.Errorf("no related task '%s' found to execute", taskrun.Spec.TaskRef.Name)
                }</span>
                <span class="cov8" title="1">newTask := deepCopyTask(task)
                updateParameters(newTask, taskrun)
                for _, step := range newTask.Spec.Steps </span><span class="cov8" title="1">{
                        c.Info("executing %s for %s", step.Name, taskrun.Metadata.Name)
                        err := c.ExecOS(p.Spec.Workspaces[0].Name+"/"+step.Workspace, step.Command[0], step.Args, true)
                        if err != nil </span><span class="cov0" title="0">{
                                return err
                        }</span>
                }
        }
        <span class="cov8" title="1">return nil</span>
}

func GenerateTaskRunFiles(path, dstPath string) error <span class="cov8" title="1">{
        bcs, err := readAllBuildConfigs(path)
        if err != nil </span><span class="cov0" title="0">{
                return err
        }</span>
        <span class="cov8" title="1">return generateTaskRunFiles(dstPath, bcs)</span>
}
</pre>
		
		</div>
	</body>
	<script>
	(function() {
		var files = document.getElementById('files');
		var visible;
		files.addEventListener('change', onChange, false);
		function select(part) {
			if (visible)
				visible.style.display = 'none';
			visible = document.getElementById(part);
			if (!visible)
				return;
			files.value = part;
			visible.style.display = 'block';
			location.hash = part;
		}
		function onChange() {
			select(files.value);
			window.scrollTo(0, 0);
		}
		if (location.hash != "") {
			select(location.hash.substr(1));
		}
		if (!visible) {
			select("file0");
		}
	})();
	</script>
</html>
