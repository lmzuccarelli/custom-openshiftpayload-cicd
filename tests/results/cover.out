mode: set
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:44.49,50.39 6 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:66.2,66.10 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:50.39,56.34 6 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:59.3,59.31 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:62.3,64.41 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:56.34,58.4 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:59.31,61.4 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:69.59,70.35 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:70.35,72.3 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:75.65,76.39 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:76.39,77.33 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:82.3,82.52 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:77.33,78.42 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:78.42,80.5 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:82.52,84.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:88.70,90.41 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:93.2,93.17 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:90.41,92.3 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:96.73,97.29 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:102.2,102.22 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:97.29,98.38 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:98.38,100.4 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:105.74,107.26 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:115.2,115.19 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:107.26,110.23 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:113.3,113.27 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:110.23,112.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:118.80,123.26 4 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:131.2,131.22 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:123.26,126.24 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:129.3,129.34 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:126.24,128.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:135.68,138.16 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:141.2,141.26 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:154.2,154.17 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:138.16,140.3 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:141.26,142.17 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:142.17,145.24 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:149.4,149.30 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:145.24,147.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:149.30,151.5 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:157.71,158.25 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:170.2,170.12 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:158.25,166.17 7 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:166.17,168.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:173.76,175.16 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:178.2,179.11 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:216.2,216.23 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:175.16,177.3 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:180.28,183.17 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:186.3,186.17 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:187.24,190.17 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:193.3,193.17 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:194.21,197.17 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:200.3,200.16 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:201.25,204.17 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:207.3,207.16 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:208.30,211.17 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:214.3,214.16 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:183.17,185.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:190.17,192.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:197.17,199.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:204.17,206.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/helpers.go:211.17,213.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:13.63,23.23 7 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:26.2,29.28 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:80.2,80.29 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:96.2,96.12 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:23.23,25.3 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:29.28,32.24 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:35.3,37.10 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:32.24,34.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:38.41,43.25 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:46.4,47.35 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:51.4,53.18 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:56.4,58.18 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:61.37,63.18 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:66.4,66.28 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:67.40,69.18 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:72.4,73.20 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:43.25,45.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:47.35,50.5 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:53.18,55.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:58.18,60.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:63.18,65.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:69.18,71.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:73.20,75.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:80.29,83.45 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:86.3,88.43 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:83.45,85.4 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:88.43,91.18 3 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:91.18,93.5 1 0
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:99.55,101.16 2 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:104.2,104.43 1 1
github.com/luigizuccarelli/custom-openshiftpayload-cicd/pkg/service/pipeline.go:101.16,103.3 1 0
